apps/meteor/app/livechat/server/lib/Contacts.ts

```diff
@@ -248,6 +257,57 @@ export async function updateContact(params: UpdateContactParams): Promise<ILivec
 	return updatedContact;
 }
 
+export async function getContactHistory(
+	params: GetContactHistoryParams,
+): Promise<PaginatedResult<{ history: VisitorSearchChatsResult[] }>> {
+	const { contactId, source, count, offset, sort } = params;
+
+	const contact = await LivechatContacts.findOneById<Pick<ILivechatContact, 'channels'>>(contactId, { projection: { channels: 1 } });
+
+	if (!contact) {
+		throw new Error('error-contact-not-found');
+	}
+
+	const visitorsIds = contact.channels?.map((channel) => channel.visitorId);
```

Comment:
Make this a `set` so you send less data to mongo in the cases where visitorId is repeated